dev-python/argon2-cffi/argon2-cffi-21.2.0.ebuild

```
@@ -0,0 +1,44 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{8..10} pypy3 )
+DISTUTILS_USE_SETUPTOOLS=pyproject.toml
+inherit distutils-r1
+
+DESCRIPTION="CFFI bindings to the Argon2 password hashing library"
+HOMEPAGE="https://github.com/hynek/argon2-cffi"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND=">=dev-python/argon2-cffi-bindings-21.2.0[${PYTHON_USEDEP}]"
+BDEPEND="
+	test? (
+		dev-python/hypothesis[${PYTHON_USEDEP}]
+	)
+"
+
+DOCS=( AUTHORS.rst CHANGELOG.md FAQ.rst README.rst )
+
+src_prepare() {
+	# Patch the tool.flit.module.name entry in pyproject.toml to include
+	# the "src" subfolder. This allows pyproject2setuppy to retrieve the
+	# dynamic version and description correctly.
+	sed -i -e 's:^name = "argon2":name = "src/argon2":' \
```

That said, ideally I'd fix pp2sp for this but honestly, I'm so tired of having to repeat every stupid hack these people come up with…

```suggestion
	sed -i -e '/^name/s:":"src/:' \
```

(untested)

Generally you can try to make sed expressions shorter when it's unlikely for them to mismatch.

---

This could be problematic: There are two `name` attributes in the `pyproject.toml`:
The one in the `[project]` section refers to the package name on PyPI and the one in `[tool.flit.module]` refers to the  name of the Python package.

This sed expression would be prefix both names with `src/`, but we only want to change the second one. I left it as is for now.

---

With regards to pp2sp: I don't think this is a hack particular to argon2-cffi. Flit seems to perform some kind of file name guessing: https://github.com/pypa/flit/blob/ec6e6b7c8994d5e79d0fd951d9e4c26a7c5d9977/tests/test_init.py#L62-L86

Do you know if other packages rely on Flit guessing the `src` prefix? If there are a bunch of them that would possibly warrant a change to pyproject2setuppy.

---

Yeah, I'm talking about flit and poetry. pp2sp already tries name inside `src/` by default but I didn't assume they would still do the guessing if name was explicitly specified.